qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -783,18 +814,36 @@ def _evaluate_plot_guess(self, plot_guess: bool) -> str:
 
     def _evaluate_function(self, fit_function: IFunction, output_workspace: str) -> str:
         """Evaluate the plot guess fit function and returns the name of the resulting guess workspace."""
+        if self.fitting_context.plot_guess_type == 'x from plot range':
+            start_x = self.current_start_x
+            end_x = self.current_end_x
+            points = 250
+        elif self.fitting_context.plot_guess_type == 'x at data points':
+            start_x = self.current_start_x
+            end_x = self.current_end_x
+            points = self.fitting_context.plot_guess_points
+        elif self.fitting_context.plot_guess_type == 'Custom x range':
+            start_x = self.fitting_context.plot_guess_start_x
+            end_x = self.fitting_context.plot_guess_end_x
+            points = self.fitting_context.plot_guess_points
+        else:
+            raise ValueError('Plot guess type: ' + self.fitting_context.plot_guess_type + ' is not recognised')
+
+        data = np.linspace(start_x, end_x, points)
+        CreateWorkspace(OutputWorkspace='__tmp_guess_workspace',
+                        DataX=data,
+                        DataY=data)
         try:
             if self._double_pulse_enabled():
                 self._evaluate_double_pulse_function(fit_function, output_workspace)
             else:
-                EvaluateFunction(InputWorkspace=self.current_dataset_name,
+                EvaluateFunction(InputWorkspace="__tmp_guess_workspace",
                                  Function=fit_function,
-                                 StartX=self.current_start_x,
-                                 EndX=self.current_end_x,
                                  OutputWorkspace=output_workspace)
         except RuntimeError:
             logger.error("Failed to plot guess.")
             return ""
+        AnalysisDataService.remove('__tmp_guess_workspace')
```

Comment:
https://github.com/mantidproject/mantid/blob/main/qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/ADSHandler/ADS_calls.py#L15

Reply:
I also think the users want to be able to "save" the result of their guess to the ADS

Reply:
I've realised that this causes a loop by triggering a signal when ADS is modified. for the moment I have stopped the removal of the tmp workspace.

Reply:
The __tmp_guess_workspace does not contain the actual guess, that is in the output of the evaluate function. this tmp workspace is just a dummy with the right x range to fit over

Reply:
ah ok, could we not just keep it internally and not add it to the ADS?